### PR TITLE
European hydrothermal plumes simulation

### PR DESCRIPTION
**Pull request by ali-ramadhan**
This PR starts an example for modeling hydrothermal plumes on a very deep ocean.

Very preliminary: at the very least we'll want #20 and maybe #191 before taking these simulations seriously, but this should be a good start.

### CODE REVIEW

**Comment by johncmarshall54** [comment]
Tides are weak because the ocean is so deep I think

**Comment by ali-ramadhan** [comment]
Ah I see. But if tidal forces are much stronger on Europa (compared to Earth) then they might matter? I don't have the numbers though.

I noticed that after ~10,000 Europan days (354 million Earth seconds) only the bottom ~10% (10 km) of the temperature field were affected by the heating flux. There are waves propagating so the vertical velocity field is non-zero almost everywhere.

Not sure if this is how things go on Europa (i.e. super long simulations will be the norm) or if we're missing key physics.

File: examples/europa_hydrothermal_plumes.jl [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Nt = Int(days*spd/dt)
+κv, νv = κ, κ  # This is also implicitly setting the Prandtl number Pr = 1.
+κh, νh = (L/H)*κv, (L/H)*νv  # Scale horizontal κ and ν by the aspect ratio.
+
+# Create the model.
+model = Model(N=(Nx, Ny, Nz), L=(Lx, Ly, Lz), νh=νh, κh=κh, νv=νv, κv=κv,
+              constants=Europa(lat=45), arch=GPU(), float_type=Float64)
+
+# Set heating flux at the bottom.
+bottom_flux = -Q / (ρ₀ * cₚ)
```
Comment (ali-ramadhan) [inline examples/europa_hydrothermal_plumes.jl]: Reminder to self: Minus sign is incorrect here as this is the bottom flux!

**Comment by ali-ramadhan** [comment]
Superseded by PR #304.

**Pull request closed without merging**